### PR TITLE
OC-927: Convert incremental ARI ingest into a standalone script

### PR DESCRIPTION
**Pull request by finlay-jisc**
The purpose of this PR was to facilitate running incremental ARI imports from an ECS service. Until now, it was only triggerable as a lambda function. This allows it to be triggered by running `npm run ariImport`.

This also brings the incremental and full ARI imports more in line with each other - both can now do dry runs, and both output the same report information (via email if triggered as a lambda, local file otherwise).

---

### Acceptance Criteria:

Incremental ARI imports are runnable as a script.

---

### Checklist:

- [x] Local manual testing conducted
- [ ] Automated tests added
- [ ] Documentation updated

---

### Tests:

API
Screenshot 2024-11-25 143443

### CODE REVIEW

File: api/package.json [hunk 1]
```diff
@@ -32,12 +32,12 @@
         "test:local": "STAGE=local jest --runInBand --testTimeout=60000",
         "test:watch": "npm run test:local -- --watch",
         "type": "tsc --noEmit",
-        "reindex": "ts-node scripts/reindex.ts",
+        "ariImport": "ts-node -r tsconfig-paths/register scripts/ariImport.ts",
```
[the remark this answers is not in the document]
Comment (finlay-jisc) [reply api/package.json]: Thanks, seems to work fine :)

File: api/package.json [hunk 2]
```diff
@@ -98,6 +97,7 @@
         "serverless-prune-plugin": "^2.1.0",
         "ts-jest": "^29.1.1",
         "ts-loader": "^9.4.4",
+        "tsx": "^4.19.2",
```
Comment (adwearing-jisc) [inline api/package.json:100]: Can ts-node be uninstalled now?

File: api/scripts/ariImport.ts [hunk 3]
```diff
@@ -1,39 +1,64 @@
 import axios from 'axios';
-import * as fs from 'fs/promises';
 import * as dotenv from 'dotenv';
-import { expand } from 'dotenv-expand';
 
 // Important to do this so that environment variables are treated the same as in deployed code.
-expand(dotenv.config());
```
Comment (adwearing-jisc) [inline api/scripts/ariImport.ts:7]: This was due to serverless right? Guessing this won't be running there anymore?

**Review by adwearing-jisc — APPROVED** [review]

File: api/package.json (hunk 2 above)
Comment (finlay-jisc) [reply api/package.json:100]: That's the weird thing; I couldn't see that it was installed in the first place!

File: api/scripts/ariImport.ts (hunk 3 above)
Comment (finlay-jisc) [reply api/scripts/ariImport.ts:7]: Yeah, from a bit of experimenting it doesn't seem to anymore. The default behaviour now is as if using dotenv, but not dotenv-expand.

**Pull request merged**